### PR TITLE
Update log4j and remove log4j-core dependency

### PR DESCRIPTION
**Pull request by DomGarguilo**
Duplicate of the changes in #3204 targeting 2.1

Removes the log4j-core dependency which was previously needed due to https://issues.apache.org/jira/browse/LOG4J2-3601 which is fixed as of version 2.20

### CODE REVIEW

**Comment by cshannon** [comment]
I don't think we need two PRs. We should be able to just have one targeting 2.1 and then when this is merged into 2.1 we can then merge it to main. The other PR, #3204, should be able to be closed I think.

File: assemble/pom.xml [hunk 1]
```diff
@@ -297,11 +297,6 @@
       <artifactId>log4j-api</artifactId>
       <optional>true</optional>
     </dependency>
-    <dependency>
-      <groupId>org.apache.logging.log4j</groupId>
-      <artifactId>log4j-core</artifactId>
-      <optional>true</optional>
-    </dependency>
     <dependency>
```
Comment (ctubbsii) [inline assemble/pom.xml:300]: We still need log4j-core in the assembly for runtime. That's why this one didn't have the same comment as the others.

**Comment by DomGarguilo** [comment]
> I don't think we need two PRs. We should be able to just have one targeting 2.1 and then when this is merged into 2.1 we can then merge it to main. The other PR, #3204, should be able to be closed I think.

Gotcha. It has been closed. I'll merge this in once it gets reviewed.

**Review by ctubbsii — CHANGES_REQUESTED** [review]

File: assemble/pom.xml (hunk 1 above)
Comment (cshannon) [reply assemble/pom.xml:300]: @ctubbsii - I was literally typing the same thing as you replied to this :)

Comment (DomGarguilo) [reply assemble/pom.xml]: Fixed in 42a4fa2

**Comment by cshannon** [comment]
@DomGarguilo - It would appear you pushed up main, I would probably just force push off 2.1 to clean it up.

**Comment by EdColeman** [comment]
Not sure what you did here - was this intended to be a merge forward?  Usually, that should not be done as a PR and depending on the base branch, be very careful on deleting the branch - you may end up deleting 2.1 from the repo.

**Comment by DomGarguilo** [comment]
> Not sure what you did here - was this intended to be a merge forward?  Usually, that should not be done as a PR and depending on the base branch, be very careful on deleting the branch - you may end up deleting 2.1 from the repo.

I accidentally merged main into this branch.

**Comment by ctubbsii** [comment]
@DomGarguilo I fixed the unintentional merge of main, merge 2.1 instead, cherry-picked your partial revert, then added another commit to fix a few things you missed. Waiting for the build to finish, but should be good after that.

**Comment by DomGarguilo** [comment]
> @DomGarguilo I fixed the unintentional merge of main, merge 2.1 instead, cherry-picked your partial revert, then added another commit to fix a few things you missed. Waiting for the build to finish, but should be good after that.

Thanks, @ctubbsii. I'll go ahead and merge it once the build completes.

**Review by ctubbsii — APPROVED** [review]

**Pull request merged**